src/backend/env/mod.rs

```diff
@@ -4843,6 +4912,46 @@ pub(crate) mod tests {
         });
     }
 
+    #[actix_rt::test]
+    async fn test_invites_with_realm() {
+        let principal = pr(4);
+        let realm_id = "realm-invite-test".to_string();
+        let invite_code = mutate(|state| {
+            // Create user with 2000 credits
+            create_user_with_credits(state, principal, 2000);
+
+            let _ = create_realm(state, principal, realm_id.clone());
+
+            assert_eq!(
+                state.create_invite(principal, 200, Some(50), Some(realm_id.clone())),
+                Ok(())
+            );
+
+            let code = invite::invites_by_principal(state, principal)
+                .first()
+                .map(|(code, _)| code.to_string())
+                .unwrap();
+
+            code
+        });
+
+        // New user should be joined to realm
+        let new_principal = pr(5);
+        assert_eq!(
+            State::create_user(new_principal, "name".to_string(), Some(invite_code)).await,
+            Ok(Some(realm_id.clone()))
+        );
+        read(|state| {
```

Comment:
Should we also assert that user has a credit balance we expect and the invite's balance decreased as expected?

Reply:
ofc, added